holoviews/core/dimension.py

```diff
@@ -116,57 +154,120 @@ class Dimension(param.Parameterized):
         instance, the string 'm' may be used represent units of meters
         and 's' to represent units of seconds.""")
 
-    values = param.ClassSelector(class_=(str, list), default=[], doc="""
-        Optional set of allowed values for the dimension that can also
-        be used to retain a categorical ordering. Setting values to
-        'initial' indicates that the values will be added during construction.""")
+    values = param.List(default=[], doc="""
+        Optional specification of the allowed value set for the
+        dimension that may also be used to retain a categorical
+        ordering.""")
 
     # Defines default formatting by type
     type_formatters = {}
     unit_format = ' ({unit})'
     presets = {} # A dictionary-like mapping name, (name,) or
                  # (name, unit) to a preset Dimension object
 
-    def __init__(self, name, **params):
+    def __init__(self, spec, **params):
         """
         Initializes the Dimension object with the given name.
         """
-        if isinstance(name, Dimension):
-            existing_params = dict(name.get_param_values())
-        elif (name, params.get('unit', None)) in self.presets.keys():
-            preset = self.presets[(str(name), str(params['unit']))]
+        if 'name' in params:
+            raise KeyError('Dimension name must only be passed as the positional argument')
+
+        if isinstance(spec, Dimension):
+            existing_params = dict(spec.get_param_values())
+        elif (spec, params.get('unit', None)) in self.presets.keys():
+            preset = self.presets[(str(spec), str(params['unit']))]
             existing_params = dict(preset.get_param_values())
-        elif name in self.presets.keys():
-            existing_params = dict(self.presets[str(name)].get_param_values())
-        elif (name,) in self.presets.keys():
-            existing_params = dict(self.presets[(str(name),)].get_param_values())
+        elif spec in self.presets.keys():
+            existing_params = dict(self.presets[str(spec)].get_param_values())
```

Comment:
This doesn't seem right, you check ``spec`` is in presets but then do the lookup with ``str(spec)``? Same below, also is ``.keys()`` needed in the conditional?